vllm/distributed/device_communicators/pynccl.py

```diff
@@ -126,6 +126,34 @@ def all_reduce(self,
                                 ncclRedOpTypeEnum.from_torch(op), self.comm,
                                 cudaStream_t(stream.cuda_stream))
 
+    def send(self, tensor: torch.Tensor, stream=None):
+        if self.disabled:
+            return
+        assert tensor.device == self.device, (
+            f"this nccl communicator is created to work on {self.device}, "
+            f"but the input tensor is on {tensor.device}")
+        if stream is None:
+            stream = self.stream
+        # Hardcoded to send to the next rank for now for simplicity.
+        dst = (self.rank + 1) % self.world_size
```

Comment:
why hardcode here?

Reply:
It's hardcoded as we only need to send to the next and prev rank for now so it's simple. We can change this if necessary though

Reply:
please don't hardcode here. just expose the general form of `send/recv`. You can add default arg.

Reply:
Changed it to expose general form with default arg `None`, which will make it next and prev rank respectively.

Reply:
Let me know if that's sufficient